src/core/server/http/http_server.ts

```diff
@@ -127,21 +127,32 @@ export class HttpServer {
     for (const router of this.registeredRouters) {
       for (const route of router.getRoutes()) {
         this.log.debug(`registering route handler for [${route.path}]`);
-        const { authRequired = true, tags } = route.options;
         // Hapi does not allow payload validation to be specified for 'head' or 'get' requests
         const validate = ['head', 'get'].includes(route.method) ? undefined : { payload: true };
+        const {
+          authRequired = true,
+          tags,
+          accepts: allow,
+          maxBytes,
+          output,
+          parse,
+        } = route.options;
         this.server.route({
           handler: route.handler,
           method: route.method,
           path: route.path,
           options: {
-            auth: authRequired ? undefined : false,
+            // Enforcing the comparison with true because plugins could overwrite the auth strategy by doing `options: { authRequired: authStrategy as any }`
+            auth: authRequired === true ? undefined : false,
             tags: tags ? Array.from(tags) : undefined,
             // TODO: This 'validate' section can be removed once the legacy platform is completely removed.
             // We are telling Hapi that NP routes can accept any payload, so that it can bypass the default
             // validation applied in ./http_tools#getServerOptions
             // (All NP routes are already required to specify their own validation in order to access the payload)
             validate,
+            payload: [allow, maxBytes, output, parse].some(v => typeof v !== 'undefined')
+              ? { allow, maxBytes, output, parse }
+              : undefined,
```

Comment:
I tried that before coming up with this solution, unfortunately, HAPI does not allow `payload` options for HEAD or GET requests (https://github.com/hapijs/hapi/blob/803f94331b13f7d1747f7f99b072d57574fd65fe/lib/route.js#L61), so setting the object, despite everything inside it is `undefined`, HAPI will reject it.

I considered filtering by method (like in https://github.com/elastic/kibana/blob/7cbea0c7f45b760415cb79e0e27a31f4ba92c24e/src/core/server/http/http_server.ts#L131) but I felt like it might cause confusion when, for whatever reason, the user decides to set any of these parameters when defining a GET route/request, leading to unexpected behaviour.

I thought it would be more convenient to go down the route of documenting it as "you can't do that" via TS (as you may have already seen in this MR) 🙃

Reply:
Ok. That answers my question about `GET` payloads.

Reply:
That was intended. I thought it was better to let them get that error vs. simply removing it for them, leading to unexpected behaviour (if the dev is explicitly setting that config, maybe he means it for something?).

Do you think we should do the validation ourselves and throw the error even earlier?

Reply:
we can add our custom validation. but as I said it's not big deal.